server/middleware/encodePassword.js

```diff
@@ -0,0 +1,16 @@
+const bcrypt = require('bcryptjs')
+require('dotenv').config()
```

Comment:
Почему бы не использовать import? Все импорты и require должны быть вверху файла

Reply:
"не использовать import" - хм... такой вопрос, пришлось погуглить.... я вообще думал что нода не умеет в ES6 import/export..... Но, оказывается, типа умеет. НО. Вот здесь нашел статью 
https://www.geeksforgeeks.org/difference-between-node-js-require-and-es6-import-and-export/
и там как - то прохладно относятся к иморту/экспорту в ноде. 
...
NOTE: You must note that you can’t use require and import at the same time in your node program and it is more preferred to use require instead of import as you are required to use experimental module flag feature to run import program.
....
Поэтому, еслине принципиально, я бы оставил в ноде везде только импорты (тем более не разводить кашу, меняя в каких - то отдельных местах.).

Reply:
Вероятно, ты меня не так понял. В файле ты используешь require, а почему бы не переписать через import и не импортировать через require? Так что, да, давай везде использовать только импорты.

Reply:
ой, сорри, оговорился, я имел в виду, конечно, что  в ноде использовал бы  только "require" (sic!), и, судя по томоу, что я нагуглил, его в динамических случаях все равно придется оставить т.к. import может быть только в безусловном начале файла.
Ну...  раз оговорился, переделаю на импорты, в конце концов интересно что получится....

Reply:
Уфф, попытался сегодня вечерком быстренько изменить, но....уві, оказалось, что  это прилично так  всего цепляет - от, естественно, замены всех "require" на  "import" (причем только дефолтный), до  ограничения версии ноды не ниже 14й,  а так же передлку динамических require на промисы  импортов, а так же замены __filename и __dirname какими-то обходными способами (они существуют только в commonJS модулях), и еще черт - знает-что  на что я еще не успел сегодня наступить.

Если задатся целью, то, думаю, всё возможно. 
Но надо ли?  Вопрос времени - еще пару потенциально рабочих вечеров на это точно потрачу..

Reply:
Так как задача большая, то давай это вынесем в отдельную задачу и заодно добавим поддержку Typescript для бека и там уже все переписывать и переделывать. Что думаешь?

Reply:
Оъединять 2 больших задачи в одну мегазадачу ))))) точно не стоит. Ладно, уже пилю импорты.  Но, ради интереса, скажи, чем импорты лучше ?

Reply:
Не стоить переписывать пол проекта в рамках данной задачи. Доделай остальные правки по пулл реквесту, а импорты и апгрейд ноды вынесем отдельно, так как очень много изменений.

Import не лучше, чем require. Просто если использовать ES6, то лучше использовать импорт в данном случае. Но так как придётся очень много переделать, то не нужно этого делать в рамках этой задачи. Рефакторинг на импорт и апгрейд ноды это самый низкий приоритет.

Доделывай остальное, к этому вернёмся позже.

Reply:
ок, откатил изменения. В целом использование импортов в ноде для меня выглядит очень сомнительно.